### ISSUE TITLE
IMAGE_RESIZE_MODE='none' not work?

### ISSUE THREAD

**Issue by world4jason** [matterport/Mask_RCNN#689]
the size of my training data is 1024x2048
and what i set in the Config class is like this

        class DataConfig(Config):
            IMAGE_RESIZE_MODE = 'none'
            IMAGE_MIN_DIM = 1024
            IMAGE_MAX_DIM = 2048

it shows that the image shape is 2048x2048
IMAGE_RESIZE_MODE              none
IMAGE_SHAPE                    [2048 2048    3]

and what in config.py is

        if self.IMAGE_RESIZE_MODE == "crop":
            self.IMAGE_SHAPE = np.array([self.IMAGE_MIN_DIM, self.IMAGE_MIN_DIM, 3])
        else:
            self.IMAGE_SHAPE = np.array([self.IMAGE_MAX_DIM, self.IMAGE_MAX_DIM, 3])

so this means i always get a square image?

=====================================

BTW is it suggested to use square images for training ?

**Comment by atlurip** [comment]
Hi I am also having the same problem.any solution for this.Not reshape the image

**Comment by Chakib-Belafdil** [comment]
Hello,

I made a small change in order to make the "none" mode work.

Changes I made in config.py:
Line 106
`# Input image resizing
    # Generally, use the "square" resizing mode for training and predicting
    # and it should work well in most cases. In this mode, images are scaled
    # up such that the small side is = IMAGE_MIN_DIM, but ensuring that the
    # scaling doesn't make the long side > IMAGE_MAX_DIM. Then the image is
    # padded with zeros to make it a square so multiple images can be put
    # in one batch.
    # Available resizing modes:
    # none:   No resizing or padding. Return the image unchanged. Set the IMAGE_WIDTH_DIM and IMAGE_HEIGHT_DIM
    # square: Resize and pad with zeros to get a square image
    #         of size [max_dim, max_dim].
    # pad64:  Pads width and height with zeros to make them multiples of 64.
    #         If IMAGE_MIN_DIM or IMAGE_MIN_SCALE are not None, then it scales
    #         up before padding. IMAGE_MAX_DIM is ignored in this mode.
    #         The multiple of 64 is needed to ensure smooth scaling of feature
    #         maps up and down the 6 levels of the FPN pyramid (2**6=64).
    # crop:   Picks random crops from the image. First, scales the image based
    #         on IMAGE_MIN_DIM and IMAGE_MIN_SCALE, then picks a random crop of
    #         size IMAGE_MIN_DIM x IMAGE_MIN_DIM. Can be used in training only.
    #         IMAGE_MAX_DIM is not used in this mode.
    IMAGE_RESIZE_MODE = "square"
    IMAGE_MIN_DIM = 800
    IMAGE_MAX_DIM = 1024

    # Only "none" mode
    IMAGE_WIDTH_DIM = 1024
    IMAGE_HEIGHT_DIM = 800`

and 

Line 218

`    def __init__(self):
        """Set values of computed attributes."""
        # Effective batch size
        self.BATCH_SIZE = self.IMAGES_PER_GPU * self.GPU_COUNT

        # Input image size
        if self.IMAGE_RESIZE_MODE == "crop":
            self.IMAGE_SHAPE = np.array([self.IMAGE_MIN_DIM, self.IMAGE_MIN_DIM,
                self.IMAGE_CHANNEL_COUNT])
        elif self.IMAGE_RESIZE_MODE == "none":
            self.IMAGE_SHAPE = np.array([self.IMAGE_WIDTH_DIM, self.IMAGE_HEIGHT_DIM,
                                         self.IMAGE_CHANNEL_COUNT])
        else:
            self.IMAGE_SHAPE = np.array([self.IMAGE_MAX_DIM, self.IMAGE_MAX_DIM,
                self.IMAGE_CHANNEL_COUNT])`
